addons/sale/models/sale.py

```diff
@@ -957,9 +966,13 @@ def write(self, values):
     customer_lead = fields.Float(
         'Delivery Lead Time', required=True, default=0.0,
         help="Number of days between the order confirmation and the shipping of the products to the customer", oldname="delay")
-    layout_category_id = fields.Many2one('sale.layout_category', string='Section')
-    layout_category_sequence = fields.Integer(string='Layout Sequence')
-    # TODO: remove layout_category_sequence in master or make it work properly
+
+    line_type = fields.Selection([
```

Comment:
- make default line_type NULL and rename (e.g. `display_type`?)
- merge constraints, and remove the purely cosmetics ones (e.g pagebreak)
- use duck-typing for skipping lines in business logic (e.g. test product_id in sales instead of line_type/display_type)